plugins/backstage-plugin-coder/src/api/CoderClient.ts

```diff
@@ -198,37 +198,39 @@ export class CoderClient implements CoderClientApi {
       request: WorkspacesRequest,
       config: CoderWorkspacesConfig,
     ): Promise<WorkspacesResponse> => {
-      const { workspaces } = await baseSdk.getWorkspaces(request);
-      const paramResults = await Promise.allSettled(
-        workspaces.map(ws =>
-          this.sdk.getWorkspaceBuildParameters(ws.latest_build.id),
-        ),
+      if (config.repoUrl === undefined) {
+        return { workspaces: [], count: 0 };
+      }
+
+      // Have to store value here so that type information doesn't degrade
+      // back to (string | undefined) inside the .map callback
+      const stringUrl = config.repoUrl;
+      const responses = await Promise.allSettled(
+        config.repoUrlParamKeys.map(key => {
+          const patchedRequest = {
+            ...request,
+            q: appendParamToQuery(request.q, key, stringUrl),
+          };
+
+          return baseSdk.getWorkspaces(patchedRequest);
+        }),
```

Comment:
This was one of the limitations I ran into with the new API endpoint: you can't make a single request for every repo key that the user adds to their workspaces config. If your config has the repo URL keys `[repo_url, alt_repo_url]`, then a request like this:
```txt
owner:me param:"repo_url=url.com" param:"alt_repo_url=url.com"
```
Won't work, because it will search for workspaces that have both keys, even though realistically, a workspace will only ever have one of them

Reply:
It's not a huge deal – the previous logic was using `Promise.allSettled`, too – but it's not making the API logic that much faster

Reply:
Yea, this is just an unfortunate consequence of our search filtering. We have to make opinions, and cannot support both union and intersection type searches _easily_ at the moment.

Our search filter is very limited at present, and there is no effort to improve it in it's current state. It's just too much of a headache is the answer atm.

Another idea I had awhile back worth peaking at: https://github.com/coder/coder/discussions/12706

Reply:
I chatted with Steven a little bit about this constraint and I guess our current search filtering cannot support both intersection and union searches. To implement both we'd basically have to duplicate all of [this logic](https://github.com/coder/coder/blob/stevenmasley%2Ftf_e2e/coderd/database/queries/workspaces.sql#L146-L387), which is already really long.

There's not a shorter syntax because we use SQLc so we must use `CASE` statements. [Here](https://github.com/coder/coder/discussions/12706) is an open discussion we can return to in the future if this ever becomes more problematic.

Anyway, all this to say I think what you have is great!